### PR TITLE
fix: Add restriction to the slo target value

### PR DESCRIPTION
**Pull request by jorgeazevedo**
## What does this change?

This makes `GuErrorBudgetAlarmExperimental` throw an error if the client passes an SLO target outside the valid range, which is 95% to 99.95%.

This is because outside of these bounds, the way the alarms are configured do not produce useful results. For example, with an SLO target of 90%, the fast alarm will never trigger (this is the one that work on an hour-long time window).

If the SLO target is overambitious, for example 99.999%, then the alarm will fire but only after the SLO target has been missed, and its error budget consumed (in an hour long window, this would happen in 26 seconds). Because this alert wouldn't be actionable, we don't recommend people use it. For this level of availability, a totally different approach needs to be taken.

## How to test

We have added unit tests to cover the different boundary conditions.

## Have we considered potential risks?
This is low risk because it's an experimental construct.

## Checklist

- [x] I have listed any breaking changes, along with a migration path [^1]
- [x] I have updated the documentation as required for the described changes [^2]

[^1]: Consider whether this is something that will mean changes to projects that have already been migrated, or to the CDK CLI tool. If changes are required, consider adding a checklist here and/or linking to related PRs.
[^2]: If you are adding a new construct or pattern, has new documentation been added? If you are amending defaults or changing behaviour, are the existing docs still valid?

### CODE REVIEW

File: src/experimental/constructs/error-budget-alarm.ts [hunk 1]
```diff
@@ -15,6 +15,16 @@ import { Topic } from "aws-cdk-lib/aws-sns";
...
 import type { GuStack } from "../../constructs/core";
 
+/**
+ * Props for creating an ErrorBudgetAlarm
+ *
+ * - sloName: The name of the SLO
+ * - sloTarget: The percentange target for the SLO, expressed as a decimal floating point number. Must be between 0.95 and 0.995
+ * - badEvents: the metric representing the bad events. Can be any IMetric type, like Metric or MathExpression
+ * - validEvents: the metric representing the events to be considered. Can be any IMetric type, like Metric or MathExpression
+ * - snsTopicNameForAlerts: the sns topic used by the alarm to send email alerts
```
Comment (jacobwinch) [inline src/experimental/constructs/error-budget-alarm.ts:25]: Minor suggestion:

```suggestion
 * - snsTopicNameForAlerts: the sns topic name (not the full arn) used by the alarm to send email alerts
```

File: src/experimental/constructs/error-budget-alarm.ts [hunk 2]
```diff
@@ -15,6 +15,16 @@ import { Topic } from "aws-cdk-lib/aws-sns";
 import { Construct } from "constructs";
 import type { GuStack } from "../../constructs/core";
 
+/**
+ * Props for creating an ErrorBudgetAlarm
```
Comment (jacobwinch) [inline src/experimental/constructs/error-budget-alarm.ts:19]: Thanks for adding these docs @jorgeazevedo & @michaelwmcnamara 💯

**Review by jacobwinch — APPROVED** [review]

File: src/experimental/constructs/error-budget-alarm.ts (hunk 2 above)
Comment (akash1810) [reply src/experimental/constructs/error-budget-alarm.ts:19]: Could we move these docs to be inlined to help [the TypeDoc site](https://guardian.github.io/cdk/)?

```ts
export interface ErrorBudgetAlarmProps {
  /**
   * The name of the SLO. Assumed to be unique per stack.
   */
  sloName: string;

  /**
   * The percentange target for the SLO, expressed as a decimal floating point number. Must be between 0.95 and 0.995.
   */
  sloTarget: number;

  badEvents: IMetric;
  validEvents: IMetric;
  snsTopicNameForAlerts: string;
}
```

File: src/experimental/constructs/error-budget-alarm.test.ts [hunk 3]
```diff
@@ -16,6 +16,32 @@ describe("The ErrorBudgetAlarmExperimental construct", () => {
...
+    const stack = simpleGuStackForTesting();
+    expect(() => {
+      new GuErrorBudgetAlarmExperimental(stack, {
+        sloName: "MapiFrontsAvailability",
+        sloTarget: 0.9996,
+        badEvents: new Metric({ metricName: "HttpErrors", namespace: "TestLoadBalancerMetrics" }),
+        validEvents: new Metric({ metricName: "HttpRequests", namespace: "TestLoadBalancerMetrics" }),
+        snsTopicNameForAlerts: "test-sns-topic",
+      });
+    }).toThrowError();
```
Comment (akash1810) [inline src/experimental/constructs/error-budget-alarm.test.ts:29]: Can we add the error message that's thrown to the test, to be sure only the expected error is seen?

File: src/experimental/constructs/error-budget-alarm.test.ts [hunk 4]
```diff
@@ -16,6 +16,32 @@ describe("The ErrorBudgetAlarmExperimental construct", () => {
...
+    const stack = simpleGuStackForTesting();
+    expect(() => {
+      new GuErrorBudgetAlarmExperimental(stack, {
+        sloName: "MapiFrontsAvailability",
+        sloTarget: 0.9499,
+        badEvents: new Metric({ metricName: "HttpErrors", namespace: "TestLoadBalancerMetrics" }),
+        validEvents: new Metric({ metricName: "HttpRequests", namespace: "TestLoadBalancerMetrics" }),
+        snsTopicNameForAlerts: "test-sns-topic",
+      });
+    }).toThrowError();
```
Comment (akash1810) [inline src/experimental/constructs/error-budget-alarm.test.ts:42]: Can we add the error message that's thrown to the test, to be sure only the expected error is seen?

File: src/experimental/constructs/error-budget-alarm.ts (hunk 2 above)
Comment (jacobwinch) [reply src/experimental/constructs/error-budget-alarm.ts:19]: Tangentially related PR: https://github.com/guardian/cdk/pull/1587!
